lib/plugins/aws/package/compile/events/iotFleetProvisioning/index.js

```diff
@@ -0,0 +1,103 @@
+'use strict';
+
+const _ = require('lodash');
+
+class AwsCompileIotFleetProvisioningEvents {
+  constructor(serverless) {
+    this.serverless = serverless;
+    this.provider = this.serverless.getProvider('aws');
+
+    this.hooks = {
+      'package:compileEvents': this.compileIotFleetProvisioningEvents.bind(this),
+    };
+
+    this.serverless.configSchemaHandler.defineFunctionEvent('aws', 'iotFleetProvisioning', {
+      type: 'object',
+      properties: {
+        enabled: { type: 'boolean' },
+        provisioningRoleArn: { $ref: '#/definitions/awsArn' },
+        templateBody: { type: 'object' },
+        templateName: { type: 'string' },
+      },
+      required: ['templateBody', 'provisioningRoleArn'],
+      additionalProperties: false,
+    });
+  }
+
+  compileIotFleetProvisioningEvents() {
+    this.serverless.service.getAllFunctions().forEach(functionName => {
+      const functionObj = this.serverless.service.getFunction(functionName);
+      let functionTemplateNumber = 0;
+
+      if (functionObj.events) {
+        functionObj.events.forEach(event => {
+          if (event.iotFleetProvisioning) {
```

Comment:
It might be slightly cleaner to not introduce extra nesting, simply by:

```javascript
if (!event.iotFleetProvisioning) return;
// event processing
```